src/sources/types.ts

```diff
@@ -47,6 +47,24 @@ export type SourceOptionalOptions = {
    * @default {@link DEFAULT_MAX_LENGTH_URL}
    */
   maxLengthURL?: number;
+
+  /**
+   * Local cache options.
+   *  * `canReadCache`: If `true`, the source will try to read from the local memory cache.
+   *  * `canStoreInCache`: If `true`, the source will store the response in the local memory cache.
+   *  * `cache`: A map of promises that are used to store the responses.
+   *
+   * If not provided, source will try to detect `CacheControl: no-cache or no-store` headers in the response and disable respective caching modes.
+   *
+   * By default, local in-memory caching is enabled
+   */
+  localCache?: LocalCacheOptions;
+};
+
+export type LocalCacheOptions = {
+  canReadCache?: boolean;
+  canStoreInCache?: boolean;
+  cache?: Map<string, Promise<unknown>>;
 };
```

Comment:
Assuming we plan to respect the definitions of HTTP `CacheControl` headers, do we want to borrow their API too? This could be...

```javascript
export type LocalCacheOptions = {
  cache?: Map<string, Promise<unknown>>;
  cacheControl?: ('no-cache' | 'no-store')[]
}
```

... which is nice because external users can find [existing API documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#directives) easily.

Reply:
Once we are down to two options, I wouldn't use a nested `cache` object and instead just provide these options directly

Reply:
Honestly, i would go in other direction.

We clearly have "domain" options (url params, post params) and meta params: cache, max url length, context, signal in near future hopefully.

I would separate those "meta" params into one object, so it's easy to pass it around between layers. Those spread operators tend to be tedious when you have to repeat them with more than handful of values in many places.
(look at old react-widgets model props).

Reply:
Fair enough